src/cdk/overlay/overlay-directives.ts

```diff
@@ -222,6 +222,9 @@ export class CdkConnectedOverlay implements OnDestroy, OnChanges {
   /** Emits when there are keyboard events that are targeted at the overlay. */
   @Output() overlayKeydown = new EventEmitter<KeyboardEvent>();
 
+  /** Emits when there are mouse events that are targeted at the overlay. */
+  @Output() overlayClick = new EventEmitter<MouseEvent>();
```

Comment:
Is this clicks _inside_ the overlay or _outside_ the overlay? If it's the latter, I would rename this to `outsideClick`

Reply:
Regarding `excludeFromOutsideClick` I was thinking for the situations where `overlayRef` has no clue who opened it, and one could not determine in the handler of `outsideClickEvents`, clicking on which element on the page should or should not detach the overlay.